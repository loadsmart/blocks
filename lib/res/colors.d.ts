declare const Colors: {
    AlgaeGreen: string;
    BrownGreyThree: string;
    Charcoal: string;
    CoralPink: string;
    DarkGrey: string;
    OrangePeel: string;
    LightGray: string;
    Tuna: string;
    VeryLightGray: string;
    White: string;
};
export default Colors;
