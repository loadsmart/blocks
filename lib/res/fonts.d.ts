declare const Fonts: {
    SharpSansBold: string;
    SharpSansItalic: string;
    SharpSansBook: string;
    SharpSansBookItalic: string;
    SharpSansExtrabold: string;
    SharpSansExtraboldItalic: string;
    SharpSansLight: string;
    SharpSansLightItalic: string;
    SharpSansMedium: string;
    SharpSansMediumItalic: string;
    SharpSansSemibold: string;
    SharpSansSemiboldItalic: string;
    SharpSansThin: string;
    SharpSansThinItalic: string;
};
export default Fonts;
