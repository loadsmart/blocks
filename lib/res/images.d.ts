declare const Images: {
    Chevron: any;
    PlusSign: any;
    SmallCheck: any;
    Warning: any;
};
export default Images;
