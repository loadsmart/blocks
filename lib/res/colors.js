"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Colors = {
    AlgaeGreen: '#14d64d',
    BrownGreyThree: '#9b9b9b',
    Charcoal: '#3c3d3a',
    CoralPink: '#fd576a',
    DarkGrey: '#666666',
    OrangePeel: '#ff9800',
    LightGray: '#d4d4d4',
    Tuna: '#484a4c',
    VeryLightGray: '#eee',
    White: '#fff',
};
exports.default = Colors;
