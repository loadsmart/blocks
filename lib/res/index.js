"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var colors_1 = __importDefault(require("./colors"));
exports.Colors = colors_1.default;
var fonts_1 = __importDefault(require("./fonts"));
exports.Fonts = fonts_1.default;
var images_1 = __importDefault(require("./images"));
exports.Images = images_1.default;
