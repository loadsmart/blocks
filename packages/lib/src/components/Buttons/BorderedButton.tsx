import React, { PureComponent } from 'react'
import {
  Image,
  ImageSourcePropType,
  ImageStyle,
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  ViewProps,
} from 'react-native'
import { Colors, Fonts } from '../../res/'

export interface BorderedButtonProps extends ViewProps {
  icon?: ImageSourcePropType
  iconStyle?: StyleProp<ImageStyle>
  title: string
  titleStyle?: StyleProp<TextStyle>
  onPress?: () => void
}

export default class BorderedButton extends PureComponent<BorderedButtonProps> {
  public render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={[styles.wrapper, this.props.style]}>
        {this.props.icon && <Image source={this.props.icon} style={this.props.iconStyle} />}
        <Text style={[styles.text, this.props.titleStyle]} numberOfLines={1}>
          {this.props.title.toUpperCase()}
        </Text>
      </TouchableOpacity>
    )
  }
}

const styles = StyleSheet.create({
  wrapper: {
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center',
    borderWidth: 1,
    borderRadius: 21,
    paddingHorizontal: 8,
    height: 42,
  },
  text: {
    fontFamily: Fonts.SharpSansBold,
    fontSize: 12,
    marginLeft: 4,
    color: Colors.Charcoal,
  },
})
