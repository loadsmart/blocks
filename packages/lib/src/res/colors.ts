const Colors = {
  AlgaeGreen: '#14d64d',
  Charcoal: '#4c4848',
  White: '#fff',
}

export default Colors
