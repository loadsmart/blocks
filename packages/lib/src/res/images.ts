const Images = {
  PlusSign: require('../../assets/images/plus-sign.png'),
}

export default Images
