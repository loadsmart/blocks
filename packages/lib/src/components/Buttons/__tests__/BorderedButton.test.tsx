import React from 'react'
import { Image, Text, TouchableOpacity } from 'react-native'
import renderer from 'react-test-renderer'
import BorderedButton, { BorderedButtonProps } from '../BorderedButton'
import { Images } from '../../../res/'

describe('BorderedButton', () => {
  let testRenderer: renderer.ReactTestRenderer
  const onPressCallback = jest.fn()

  it('executes onPress callback', () => {
    const props: BorderedButtonProps = {
      title: 'Call Loadsmart',
      onPress: onPressCallback,
    }
    testRenderer = renderer.create(<BorderedButton {...props} />)

    const touchable = testRenderer.root.findByType(TouchableOpacity)
    touchable.props.onPress()
    expect(onPressCallback).toHaveBeenCalledTimes(1)
  })

  describe('when icon is undefined', () => {
    beforeEach(() => {
      const props: BorderedButtonProps = {
        title: 'Call Loadsmart',
      }
      testRenderer = renderer.create(<BorderedButton {...props} />)
    })

    it('renders title but not icon', () => {
      expect(testRenderer.root.findAllByType(Text)).toHaveLength(1)
      expect(testRenderer.root.findAllByType(Image)).toHaveLength(0)
    })

    it('has a valid snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })

  describe('when icon is set', () => {
    beforeEach(() => {
      const props: BorderedButtonProps = {
        icon: Images.PlusSign,
        title: 'Call Loadsmart',
      }
      testRenderer = renderer.create(<BorderedButton {...props} />)
    })

    it('renders both icon and title', () => {
      expect(testRenderer.root.findAllByType(Text)).toHaveLength(1)
      expect(testRenderer.root.findAllByType(Image)).toHaveLength(1)
    })

    it('has a valid snapshot', () => {
      expect(testRenderer.toJSON()).toMatchSnapshot()
    })
  })
})
