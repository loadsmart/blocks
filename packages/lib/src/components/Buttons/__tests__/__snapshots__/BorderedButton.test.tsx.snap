// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BorderedButton when icon is set has a valid snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 21,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/plus-sign.png",
      }
    }
  />
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#4c4848",
          "fontFamily": "SharpSans-Bold",
          "fontSize": 12,
          "marginLeft": 4,
        },
        undefined,
      ]
    }
  >
    CALL LOADSMART
  </Text>
</View>
`;

exports[`BorderedButton when icon is undefined has a valid snapshot 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 21,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 8,
    }
  }
>
  <Text
    numberOfLines={1}
    style={
      Array [
        Object {
          "color": "#4c4848",
          "fontFamily": "SharpSans-Bold",
          "fontSize": 12,
          "marginLeft": 4,
        },
        undefined,
      ]
    }
  >
    CALL LOADSMART
  </Text>
</View>
`;
